### ISSUE TITLE
Consider using IP2Location.io API as a failover

### ISSUE THREAD

**Issue by ip2location** [datawookie/medusa-proxy#6]
Currently, you're calling ip-api for the geolocation. As a failover, please consider using the IP2Location.io API via the endpoint https://api.ip2location.io/.

**Comment by datawookie** [comment]
Hi @ip2location,

Thanks for the note. This would be a good option. However, based on what I see on your site I would need to have a token to use the API. In principle that's not an issue. If you'd like to submit a PR with the proposed changes then please go ahead.

Thanks, Andrew.

**Comment by ip2location** [comment]
Hi @datawookie,

We do have a keyless API, you can use this API like this:

```bash
curl "https://api.ip2location.io/?ip=8.8.8.8&format=json"
```
For more information on our keyless API, you can kindly visit the "FAQs" section in our [pricing page](https://www.ip2location.io/pricing).

**Comment by datawookie** [comment]
Hi @ip2location, aha! But there are only 500 queries allowed per day. That would not work for Medusa since it would burn through that quickly!

**Comment by ip2location** [comment]
@datawookie What is the limit of existing implementation?

Based on ip-api web site, it has usage limit of 45 requests per minute before temporarily blocked. Repeatedly exceeding the limit will result in banned for up to 1 hour.

**Comment by datawookie** [comment]
Yes, and that limit is fine for Medusa since it's per minute and Medusa does not (generally) exceed that rate.

A limit of 500 requests per day is completely insufficient.

**Comment by abdullahdevrel** [comment]
@datawookie You can explore IPinfo.io if you want. Allow me to first address our advantage over IP-api.com.

First, we have better data. We use a latency-based geolocation methodology over the traditional ISP-reported data (WHOIS/Geofeed). We operate 700 servers worldwide just to have this edge. Feel free to explore this idea on [our blog post](https://ipinfo.io/blog/probe-network-how-we-make-sure-our-data-is-accurate/) and [community](https://community.ipinfo.io/tag/accuracy).

We support SSL/HTTPS even without a token. Your current provider does not have that for tokenless access.

---

Now, in addition to our current proposal, we have plenty more to offer here on this current proposal:

- Tokenless access to 1,000 requests per day
- Free tier token access to 50,000 requests per month.

That's it. I do not even know if we have rate limits or something, to be honest. We processed 2 trillion requests last year, which would probably indicate we do not have a rate limit. I believe you can use our [bulk API](https://ipinfo.io/blog/bulk-ip-lookups/) and consume your entire month's allocated limit in one go.

Usage example:

```
curl https://ipinfo.io/8.8.8.8
```

```
{
    "ip": "8.8.8.8",
    "hostname": "dns.google",
    "anycast": true,
    "city": "Mountain View",
    "region": "California",
    "country": "US",
    "loc": "37.4056,-122.0775",
    "org": "AS15169 Google LLC",
    "postal": "94043",
    "timezone": "America/Los_Angeles"
}
```

Let me know what you think. Thanks!
